core/src/main/java/org/apache/accumulo/core/rpc/TraceProtocolFactory.java

```diff
@@ -39,17 +40,19 @@
   public TProtocol getProtocol(TTransport trans) {
     return new TCompactProtocol(trans) {
       private Span span = null;
+      private Scope scope = null;
 
       @Override
       public void writeMessageBegin(TMessage message) throws TException {
-        span = TraceUtil.createSpan(this.getClass(), "client:" + message.name, SpanKind.CLIENT);
-        span.makeCurrent();
+        span = TraceUtil.createSpan(this.getClass(), message.name, SpanKind.CLIENT);
+        scope = span.makeCurrent();
         super.writeMessageBegin(message);
       }
 
       @Override
       public void writeMessageEnd() throws TException {
         super.writeMessageEnd();
+        scope.close();
         span.end();
```

Comment:
Does this order matter?

```suggestion
        span.end();
        scope.close();
```

Since `span.end()` typically updates the stop time for a span, and I suspect that `scope.close()` may start triggering exporters, it may be useful to end the span before closing the scope.

Reply:
The normal pattern is that the scope get closed first, then the span:
```
    Span span = TraceUtil.createSpan(Merge.class, "start", SpanKind.CLIENT);
    try (Scope scope = span.makeCurrent()) {
      ...
    } catch (Exception ex) {
      TraceUtil.setException(span, ex, true);
      throw new MergeException(ex);
    } finally {
      span.end();
    }
```